collector/kernel/buffered_poller.cc

```diff
@@ -183,9 +193,21 @@ void BufferedPoller::process_samples(bool is_event)
 
       /* unknown message -- this is a bug */
       throw std::runtime_error("unexpected bpf message\n");
-    } else if (reader.peek_type() == PERF_RECORD_LOST) {
-      lost_count_ += reader.peek_n_lost();
-      reader.pop();
+    } else if (peek_type == PERF_RECORD_LOST) {
+      if (debug_bpf_lost_samples_tmp) {
+        debug_bpf_lost_samples_ = false;
+        lost_count_ += 1;
+      } else {
+        lost_count_ += reader.peek_n_lost();
+        reader.pop();
+      }
+
+      send_report_if_recent_loss();
+
+      log_.warn("Lost {} bpf samples - restarting kernel_collector.", lost_count_);
+      kernel_collector_restarter_->request_restart();
```

Comment:
When lost BPF samples are detected, send_report_if_recent_loss() will be called immediately to notify the pipeline server about the lost samples, followed by a request to restart the KernelCollector.